#BE-4163 Docs - Testing Distribution Rework (#582)
appcircleio/appcircle-docusaurus: docs/distribute/create-or-select-a-distribution-profile.md

When you build and distribute an application with multiple flavors, and `.apk` file will be created for each flavor. When the build is distributed, all of the binaries will be seen on the distribution profile:

<Screenshot url="https://cdn.appcircle.io/docs/assets/BE-4163-main9.png" />

#### How to see the multiple flavor results

If you also want to download or see the output, you can check through the following steps within the Build Profile:

- Click the three dot under the actions tab
- Click **Download Artifacts** to see all the build outputs.

<Screenshot url='https://cdn.appcircle.io/docs/assets/BE-4163-main10.png' />

<Screenshot url="https://cdn.appcircle.io/docs/assets/BE-4163-main11.png" />

:::info

If your Git commit has any messages, they will be included in the distribution in Message To Testers area.

:::

### Upload using API & CLI

If you use your own CI structre, you can use our Appcircle API & CLI to upload binaries to your Distribution Profile.

To get more information, please refer to our [API & CLI](/appcircle-api) documentation.

## Settings

> You can customize the settings of your distribution profile. Click on the distribution profile and then the settings button within the profile.

Reviewer: ```suggestion
The settings of your distribution profile can be customized. Click on the distribution profile, then click the settings button within the profile.
```

Your application will be sent to the related testing groups as soon as your build is complete, or when a package is manually uploaded or deployed via CLI.

### Authentication

Under the Authentication tab in the settings, you can select a preferred authentication method for sharing your application. This will be the login method for the [Tester Portal](/distribute/downloading-binaries).

- **None**: No authentication, anyone with the link can download binary files
- **Static Username and Password**: One single username and password for all testers
- **SSO Login**: SSO login for all testers (Enterprise accounts only)
- **LDAP Login**: LDAP login for all testers (Enterprise accounts only)

<Screenshot url='https://cdn.appcircle.io/docs/assets/BE-4163-main4.png' />

To add your SSO and LDAP details, go to [My Organization](/account/my-organization) Integrations screen and press the "Connect" button next to SSO Login or LDAP Login under the "Connections" section.

<ContentRef url="/account/my-organization/sso-providers-configuration/single-sign-on">SSO Login</ContentRef>
<ContentRef url="/account/my-organization/ldap-login">LDAP Login</ContentRef>

:::info

> If your organization does not have SSO and LDAP details configured, these authentication methods will not be visible in your Distribution Profile settings.

Reviewer: ```suggestion
If SSO and LDAP details are not configured for your organization, these authentication methods will not be visible in the Distribution Profile settings.
```

:::

### Device Registration (iOS Only)

You may enable this option to automatically register devices in the Apple Developer Portal and update your Ad Hoc provisioning profiles. You must select App Store Connect API Key, a build profile, and a configuration.

<Screenshot url='https://cdn.appcircle.io/docs/assets/BE-4163-main5.png' />

### Public Link

You may enable a public link for your distribution. This allows anyone who has the link to access all artifacts of the distribution profile. Additionally, users can now conveniently scan a QR code to retrieve the public link directly. This simplifies the process of accessing and sharing the distribution link, making it more accessible for users on mobile devices or others who prefer quick scanning.

<Screenshot url='https://cdn.appcircle.io/docs/assets/BE-4163-main6.png' />

:::info

The Tester Portal that you will have access via the Public Link, will have the same authentication method that you have set from the authentication settings.

:::

## Share Binary

### Share your application with the test groups manually

> When you have your build ready or uploaded your binary file to Appcircle, you can manually send the file to testers to let them download, install on their devices and run the application for testing purposes.

Reviewer: ```suggestion
Once your build is ready or the binary file is uploaded to Appcircle, the file can be manually sent to testers for downloading, installing on their devices, and running the application for testing purposes.
```

> Click on the Share with Testers button and select the [testing groups](/distribute/testing-groups) you created previously that need to receive this version of your application. Alternatively, you can enter email addresses of the testers here to send them the application regardless of the testing groups.

Reviewer: ```suggestion
Click on the 'Share with Testers' button, and the [testing groups](/distribute/testing-groups) previously created can be selected to receive this version of your application. Alternatively, email addresses of testers can be entered here to send the application directly, bypassing the testing groups.
```

You can also add a message to testers including testing instructions and release notes.

<Screenshot url='https://cdn.appcircle.io/docs/assets/BE-4163-main14.png' />

You can automate this message using [Release Notes Component](https://github.com/appcircleio/appcircle-release-notes-component/). You can enrich the contents of your release notes with environment variables or Ruby snippets. The following default template will print the branch name, commit hash and commit message.

```ruby
Branch: $AC_GIT_BRANCH
Commit Hash:  <%= ENV['AC_GIT_COMMIT'][0..6] %>
Commit Message: $AC_COMMIT_MESSAGE
```

<Screenshot url='https://cdn.appcircle.io/docs/assets/BE-4163-mail.png' />

:::tip

> Once you share your app versions with your testers, you can see the most recent sharing time on your testing distribution profile card.

Reviewer: ```suggestion
After sharing your app versions with testers, the most recent sharing time will be displayed on your testing distribution profile card.
```

:::

:::info

> If you are using Self Hosted version of Appcircle, instead of the default noreply@appcircle.io, you can use your own business domain for the distribution emails.

Reviewer: ```suggestion
If you are using the Self-Hosted version of Appcircle, you can configure it to use your own business domain for distribution emails instead of the default noreply@appcircle.io.
```
